src/main/java/reposense/RepoSense.java

```diff
@@ -85,26 +85,37 @@ public static void main(String[] args) {
             RepoConfiguration.setIsFindingPreviousAuthorsPerformedToRepoConfigs(configs,
                     cliArguments.isFindingPreviousAuthorsPerformed());
 
-            if (RepoConfiguration.isAnyRepoFindingPreviousAuthors(configs)
-                    && !GitVersion.isGitVersionSufficientForFindingPreviousAuthors()) {
+            boolean isGitVersionInsufficient = RepoConfiguration.isAnyRepoFindingPreviousAuthors(configs)
+                    && !GitVersion.isGitVersionSufficientForFindingPreviousAuthors();
+            if (isGitVersionInsufficient) {
                 logger.warning(FINDING_PREVIOUS_AUTHORS_INVALID_VERSION_WARNING_MESSAGE);
                 RepoConfiguration.setToFalseIsFindingPreviousAuthorsPerformedToRepoConfigs(configs);
             }
 
+            boolean isTestMode = cliArguments.isTestMode();
+
+            if (isTestMode) {
+                assert !isGitVersionInsufficient;
+                AuthorConfiguration.setHasAuthorConfigFile(false);
+            }
+
             List<Path> reportFoldersAndFiles = ReportGenerator.generateReposReport(configs,
                     cliArguments.getOutputFilePath().toAbsolutePath().toString(),
                     cliArguments.getAssetsFilePath().toAbsolutePath().toString(), reportConfig,
                     formatter.format(ZonedDateTime.now(cliArguments.getZoneId())),
                     cliArguments.getSinceDate(), cliArguments.getUntilDate(),
                     cliArguments.isSinceDateProvided(), cliArguments.isUntilDateProvided(),
                     cliArguments.getNumCloningThreads(), cliArguments.getNumAnalysisThreads(),
-                    TimeUtil::getElapsedTime, cliArguments.getZoneId());
-            FileUtil.zipFoldersAndFiles(reportFoldersAndFiles, cliArguments.getOutputFilePath().toAbsolutePath(),
-                    ".json");
+                    TimeUtil::getElapsedTime, cliArguments.getZoneId(), cliArguments.isFreshClonePerformed());
+
+            if (!isTestMode) {
+                FileUtil.zipFoldersAndFiles(reportFoldersAndFiles, cliArguments.getOutputFilePath().toAbsolutePath(),
+                        ".json");
+            }
 
             logger.info(TimeUtil.getElapsedTimeMessage());
 
-            if (cliArguments.isAutomaticallyLaunching()) {
+            if (!isTestMode && cliArguments.isAutomaticallyLaunching()) {
```

Comment:
Can I know what's the reason for adding this? I don't think it's necessary to enforce this test code-related behavior in production code

Reply:
I think I added it to all the codes that previously are not in the old `runTest()`. This also applies to [`FileUtil.zipFoldersAndFiles(...)`](https://github.com/reposense/RepoSense/pull/1816/files/f58870a6305b843c8f896af16b142ff450bc25a4#diff-c782c38fa12442cbc9d82ce735beb26380383527acd4255637c2905ff012ea7bR111).

Reply:
does it still work if those two `isTestCode` segments are removed? This and the zipFoldersAndFiles. My original intention was to actually get rid of `generateReposReport`'s own way of running the report generation so I don't think it's important to retain this behavior unless it does not work

Reply:
Just tested locally, can still pass all test cases when they are removed. Negligible difference in run time also. I shall remove them then.